src/rename.rs

```diff
@@ -35,14 +57,37 @@ impl RenameProposal {
         }
     }
 
-    /// Execute this proposal and rename the `src` to the `dst`.
-    pub fn rename(&self) -> Result<()> {
-        fs::rename(&self.src, &self.dst).map_err(|e| {
-            format_err!(
-                "error renaming '{}' to '{}': {}",
-                self.src.display(), self.dst.display(), e,
-            )
-        })?;
+    /// Execute this proposal according to `RenameAction`.
+    pub fn rename(&self, action: &RenameAction) -> Result<()> {
+        match action {
+            RenameAction::Rename => {
+                fs::rename(&self.src, &self.dst).map_err(|e| {
+                format_err!(
+                    "error renaming '{}' to '{}': {}",
+                    self.src.display(), self.dst.display(), e,
+                )})?;
+            }
+            RenameAction::Symlink => {
+                #[cfg(unix)] {
+                    use std::os::unix;
+                    unix::fs::symlink(&self.src, &self.dst).map_err(|e| {
+                        format_err!(
+                            "error symlinking '{}' to '{}': {}",
+                            self.src.display(),
+                            self.dst.display(),
+                            e,
+                        )
+                    })?;
+                }
```

Comment:
Could you return an error for `#[cfg(not(unix))]`? I know this path isn't exercised on `not(unix)`, but that might not always be true in the future. Otherwise, this would just silently return success without actually doing anything.

Reply:
:+1: